packages/gatsby-source-drupal/src/utils.js

```diff
@@ -110,6 +110,68 @@ const handleReferences = (
 
 exports.handleReferences = handleReferences
 
+const handleDeletedNode = async ({
+  actions,
+  node,
+  getNode,
+  createNodeId,
+  entityReferenceRevisions,
+}) => {
+  const deletedNode = getNode(
+    createNodeId(
+      createNodeIdWithVersion(
+        node.id,
+        node.type,
+        getOptions().languageConfig ? node.attributes.langcode : `und`,
+        node.attributes?.drupal_internal__revision_id,
+        entityReferenceRevisions
+      )
+    )
+  )
+
+  // Remove the deleted node from backRefsNamesLookup
+  backRefsNamesLookup.delete(deletedNode)
+
+  // Remove relationships from other nodes and re-create them.
+  Object.keys(deletedNode.relationships).forEach(key => {
+    let ids = deletedNode.relationships[key]
+    ids = _.isArray(ids) ? ids : [ids]
+    ids.forEach(id => {
+      const node = getNode(id)
+      let referencedNodes = referencedNodesLookup.get(node)
+      if (referencedNodes.includes(deletedNode.id)) {
+        // Loop over relationships and cleanup references.
+        Object.entries(node.relationships).forEach(([key, value]) => {
+          // If a string ref matches, delete it.
+          if (_.isString(value) && value === deletedNode.id) {
+            delete node.relationships[key]
+          }
+
+          // If it's an array, filter, then check if the array is empty and then delete
+          // if so
+          if (_.isArray(value)) {
+            value = value.filter(v => v !== deletedNode.id)
+
+            if (value.length === 0) {
+              delete node.relationships[key]
+            } else {
+              node.relationships[key] = value
+            }
+          }
+        })
+
+        // Remove deleted node from array of referencedNodes
+        referencedNodes = referencedNodes.filter(nId => nId !== deletedNode.id)
+        referencedNodesLookup.set(node, referencedNodes)
+      }
+    })
```

Comment:
I'm a bit lost here, shouldn't we also call createNode to re-create the nodes and deleteNode? Now we're mutating node.relationships without saving it, wouldn't it cause problems?

Reply:
Oh right, yeah forgot that part 🙃